### PR TITLE
Feat: support vmss tools

### PR DESCRIPTION
**Pull request by gossion**
This pull request introduces Virtual Machine Scale Set (VMSS) support for Azure clients, along with new tools and handlers to manage VMSS resources efficiently. The changes include updates to the Azure client to handle VMSS operations, new utility functions for parameter extraction, and the addition of tools and handlers for interacting with VMSS resources.

### Azure Client Enhancements:
* Added support for VMSS clients (`VMSSClient` and `VMSSVMsClient`) in the `SubscriptionClients` struct to enable VMSS-related operations. (`internal/azureclient/client.go`, [internal/azureclient/client.goR26-R27](diffhunk://#diff-c1ff5094a99a88a9166175c4cfd96ea7c2d33cc66ee9f2d2f6dc70735eeeb286R26-R27))
* Implemented the `GetVMSS` method to retrieve VMSS information using caching for improved performance. (`internal/azureclient/client.go`, [internal/azureclient/client.goR309-R337](diffhunk://#diff-c1ff5094a99a88a9166175c4cfd96ea7c2d33cc66ee9f2d2f6dc70735eeeb286R309-R337))
* Extended the `GetResourceByID` method to support VMSS resource types. (`internal/azureclient/client.go`, [internal/azureclient/client.goR367-R368](diffhunk://#diff-c1ff5094a99a88a9166175c4cfd96ea7c2d33cc66ee9f2d2f6dc70735eeeb286R367-R368))

### Utility Functions:
* Added the `ExtractAKSParameters` function to validate and extract common AKS parameters from input maps. (`internal/components/common/helpers.go`, [internal/components/common/helpers.goR1-R36](diffhunk://#diff-1ba3f9ba23f1a55957709b50e870279ce4fa3943a75ed80cb54e7beb05220f3eR1-R36))

### VMSS Tools and Handlers:
* Introduced tools for VMSS operations, including `get_vmss_info_by_node_pool` and `get_all_vmss_by_cluster`, to fetch detailed VMSS configurations. (`internal/components/compute/registry.go`, [internal/components/compute/registry.goR1-R73](diffhunk://#diff-e762edebb6f72ea0239b83cc23034d963b2c7a7408f531502f26d1e08221e5b7R1-R73))
* Implemented handlers for VMSS commands to retrieve VMSS information by node pool and for all node pools in a cluster. (`internal/components/compute/handlers.go`, [internal/components/compute/handlers.goR1-R177](diffhunk://#diff-8e28420547dc2ecc34a3e4850b0e59bece91111adabf3c76a7216795e840eb88R1-R177))

### Testing:
* Added unit tests for `ExtractAKSParameters` and VMSS handlers to ensure functionality and validate edge cases. (`internal/components/common/helpers_test.go`, [[1]](diffhunk://#diff-4dc6ecb6d778a7c951277bf83135925f1ffe1ca3f1356a7252552d5a0d39e18eR1-R99); `internal/components/compute/handlers_test.go`, [[2]](diffhunk://#diff-8e7c9e396c7b03716e570ec0d4bc0f3ca2d5dd77af669bdccf29056b9c703431R1-R120)

These changes significantly enhance the Azure client and tooling capabilities, enabling more efficient management of VMSS resources in AKS environments.

### CODE REVIEW

**Comment by nilo19** [comment]
Do we have prompt to restrict llm to only call vmss tools when the cluster is vmss typed?

**Comment by thomas1206** [comment]
why we change the files in detector and fleet folder?

File: internal/components/compute/registry.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		mcp.WithString("resource_group",
+			mcp.Description("Azure Resource Group containing the AKS cluster"),
+			mcp.Required(),
+		),
+		mcp.WithString("cluster_name",
+			mcp.Description("Name of the AKS cluster"),
+			mcp.Required(),
+		),
+		mcp.WithString("node_pool_name",
+			mcp.Description("Name of the node pool to get VMSS information for"),
```
Comment (feiskyer) [inline internal/components/compute/registry.go:27]: how about combine the following get_all_vmss_by_cluster together here, where empty node_pool_name means all nodepools?

**Review by feiskyer — COMMENTED** [review]
PR needs a rebase due to another change merged

**Review by feiskyer — APPROVED** [review]
LGTM
